lib/reek/smells/nil_check.rb

```diff
@@ -30,7 +30,7 @@ def examine_context(ctx)
       # A base class that allows to work on all nodes of a certain type.
       #
       class NodeFinder
-        def initialize(ctx, type, detector = nil)
+        def initialize(ctx, type, detector: nil)
```

Comment:
I'd make detector a required argument instead. The case where detector is nil blows up anyway.